### ISSUE TITLE
Network Response Time Out Error

### ISSUE THREAD

**Issue by aayanqazi** [react-community/create-react-native-app#144]
Uncaught Error: Java.net,sockettimeoutException: failed to connect to after 10000ms in android after scan QR code

**Comment by dikaiosune** [comment]
Hi! Without the information from the new issue template it's very difficult to help you debug this. In general, you want to make sure that you can reach the development server from your phone. If you take the URL that CRNA prints at startup, and replace `exp://` with `http://`, you should be able to reach that in your phone's browser and see some JSON. If you can see that JSON, but scanning the QR code doesn't work, please open a new issue and complete all of the required sections of the new issue template.

**Comment by yash2code** [comment]
same issue
![whatsapp image 2017-04-24 at 13 14 44](https://cloud.githubusercontent.com/assets/20199248/25327206/e692724e-28f0-11e7-8771-90afe969d884.jpeg)

**Comment by aayanqazi** [comment]
@yash2code  first check your ip address of your pc .... write <code>ipconfig</code> on cmd ... and then write <code>set REACT_NATIVE_PACKAGER_HOSTNAME='my-custom-ip-address-or-hostname' </code> then start your project again ...

**Comment by yash2code** [comment]
@aayanqazi 

>  Ethernet adapter VirtualBox Host-Only Network #2: IPv4 Address. . . . . . . . . . . : 192.168.56.1

Now, what to write in place of  `'my-custom-ip-address-or-hostname'`

**Comment by aayanqazi** [comment]
@yash2code Your ipv4 address that found in your ipconfig

**Comment by yash2code** [comment]
@aayanqazi  tried that but didnt worked :(

**Comment by yash2code** [comment]
@dikaiosune  please re open this issue

**Comment by steadycoding** [comment]
Review this [https://github.com/react-community/create-react-native-app/blob/master/react-native-scripts/template/README.md#networking](url)

**Comment by aayanqazi** [comment]
@yash2code  can u send screenshot of <code>ipconfig</code> command ?

**Comment by yash2code** [comment]
@steadycoding  url timeout in phone's chrome browser but works in desktop chrome browser. Using hotspot as said.

**Comment by yash2code** [comment]
@aayanqazi  my screenshots of` ipconfig`


![capture](https://cloud.githubusercontent.com/assets/20199248/25367908/95e3f8bc-2996-11e7-9972-776aecfbef0a.PNG)
![capture1](https://cloud.githubusercontent.com/assets/20199248/25367907/95d4ca36-2996-11e7-87ad-29cfb28e23a0.PNG)

**Comment by aayanqazi** [comment]
Use Wireless LAN Adapter WI-FI Ipv4 Address 192.168.43.244 @yash2code

**Comment by yash2code** [comment]
@aayanqazi  when I goto exp://192.168.43.244 , it starts fetching js bindle but app doesnt opens and restarts my phone giving error `this.stream.clearLine() is not a function` , Also , on scanning qr code gives same error: `Network response time out error`

**Comment by yash2code** [comment]
@dikaiosune  I think issue needs to re open .

**Comment by Jitender-Rathore** [comment]
i am getting the same issue as @yash2code
Have done all the steps which you mentioned but still the problem persists.
My PC and phone are both connected to wifi. In browser , if i hit the ip address i am getting the json but in phone its giving timeout error. 
Here is the screenshot of the ipconfig
![ipconfig](https://cloud.githubusercontent.com/assets/27485960/26529241/20756544-43da-11e7-858c-8093ac9ecaaf.png)

**Comment by yash2code** [comment]
@Jitender-Rathore  send the full screenshot of ipconfig

**Comment by Jitender-Rathore** [comment]
the sccreenshot which i upload is a full one . 
I also tried with hotspot .This time i was getting the json when i enter the ip address in mobile browser and desktop but expo was showing the same error

**Comment by yash2code** [comment]
from my screenshot ........i used this 192.168.43.244  and it worked

**Comment by phamios** [comment]
how to fix ? I got the same problem

**Comment by yash2code** [comment]
@phamios  show the output of ipconfig

**Comment by CyrisXD** [comment]
This issue is caused by having virtual network devices, usually installed by virtual machines.
React Native chooses the first available network IP which may not be the correct one.

To fix this issue you have to find your local IP address, by entering the Command Prompt and typing: 
`ipconfig` 
or
`ipconfig /all`

If you are on WiFi, look for the WiFi section, and note down your **IPv4 Address**
If you are on Ethernet, look for the Ethernet Adapter section, and note down your **IPv4 Address**

For Example: 

![untitled](https://user-images.githubusercontent.com/3248638/27128409-735c21fa-5152-11e7-9e2f-5e4276fd843e.jpg)

Once you have your local IP address, navigate to your project folder inside Command Prompt.
Inside your project folder run the command.

`REACT_NATIVE_PACKAGER_HOSTNAME=my-custom-ip-address`

Replace **my-custom-ip-address** with your **IPv4 Address** you noted down earlier.

For example:

![cmd_2017-06-14_22-42-00](https://user-images.githubusercontent.com/3248638/27128516-e9dda04c-5152-11e7-8351-a111d8fac4c1.png)

Then run `npm start`
Scan your QRCode through the Expo app and it should be working.

**Comment by mkmpvtltd1** [comment]
https://stackoverflow.com/a/44600431/1756068

**Comment by BastienGermain** [comment]
Hello,

I have the same issue.

I try this : set REACT_NATIVE_PACKAGER_HOSTNAME=my-custom-ip-address
And also the solution from Stack Overflow (open the port 19000 from the machine)

It still doesn't work. 

Thanks for your help

**Comment by lcruz4** [comment]
I am using git bash and had to use
export REACT_NATIVE_PACKAGER_HOSTNAME='192.168.1.123'
instead of set REACT_NATIVE_PACKAGER_HOSTNAME
still having issues but got past the wrong ip issue.

FYI when you run npm start it tells you in the output the IP it's using. If that IP is not the IP you want then the REACT_NATIVE_PACKAGER_HOSTNAME environment variable is not set properly

**Comment by hrastnik** [comment]
I've had the same issue. The problem is that the command `yarn start` or `npm start` finds the first enabled network adapter it finds which just happens to be the adapter for VirtualBox on my (and many other machines). The solution is to either:
 - manually set REACT_NATIVE_PACKAGER_HOSTNAME environment variable, or
 - disable the unused network adapter in Windows' Network Connection (see image)

The problem is that Expo finds the first Network adapter, and not the active network adapter that provides the internet connection.

![untitled](https://user-images.githubusercontent.com/3407234/28021297-b1e1b22a-6587-11e7-9db4-152b9f1b057c.png)

**Comment by reza7rm** [comment]
I tested all solutions but Expo still encounters timeout error, it's really on the nerve! deploying using react-native cli was much much better and easier, now I donno why even cli has some issues with watchman and changing the files doesn't reflect on device.

Windows 10
react native 0.47

**Comment by aliirezaaa** [comment]
i used below command and scan QR with expo app and its work !!
`npm start`
but when i use open on android , still has time out error .

**Comment by miguelalche** [comment]
@CyrisXD answer did it for me!! check it out! 

I went around disabling virtualhoust, firewalls, etc... but what really happened is that It was exporting the wrong ip adress.

Remember that if you are on windows you need to do "export REACT_NATIVE_PACKAGER_HOSTNAME=my-custom-ip-address" instead of "set REACT_NATIVE_PACKAGER_HOSTNAME=my-custom-ip-address"

**Comment by CristhianUNSa** [comment]
This worked for me: I've downloaded adb from android tools then `adb reverse tcp:19000 tcp:19000`, and I didn't put IPv4 address as recommended... my localhost is 127.0.0.1, so manually entering exp://127.0.0.1:19000 in Expo App worked for me. I don't know why is this, just trial and error

**Comment by anass05** [comment]
for anyone who's still facing this problem on windows 
1) make sure your phone is connected to the same network as you computer 
2) make sure that **windows firewall** isn't blocking that port, if so make an exception for that port (1900 in my case)

**Comment by MichaelJob** [comment]
My solution I found while struggeling with the same problem on another machine (osx sierra), while it was working on my first one  (osx sierra) - McAfee was blocking it (which was only installed on the second one); disabled it's Firewall and voilà, it works (ios/android)

**Comment by sustained** [comment]
I'm having the same issue although my setup is really weird and maybe it's just not possible to get this working.

I have a USB broadband dongle and my internet is via a mobile network (4G) plus I'm on Linux (Fedora) and well these devices are not known to work amazingly on Linux to begin with. I reckon that the ISP may well be blocking ports, though, or otherwise interfering with traffic in some way, or something like that (UK mobile networks are notorious for this).

I've tried adding the port to the firewall and so on and so forth (read through the entire thread) but to no avail. Can anyone try to help me?

**Comment by CristhianUNSa** [comment]
@sustained Have you tried the solution I posted before? Your phone doesn't need to be on the same network, as it will be connected via USB port (You have to enable USB Debugging on your phone, this is for Android devices). You will need to download `adb` from [Android SDK Platform Tools](https://developer.android.com/studio/releases/platform-tools.html)

**Comment by sustained** [comment]
I'm actually happy to report that I got it working by opening ports 19000, 19001 and 19002!

**Comment by lulzimfazlija** [comment]
@hrastnik had it right, works on my windows machine by disabling all other network adapters, as mentioned by hrastnik, expo seems not quite smart when detecting the correct network adapter, not looking for the active one, but instead the first one... So disable all other network adapters :)

**Comment by svekl** [comment]
If you have several network connections, but can't disable any of them, here is nice solution how to do it on Windows machine
`<spoiler>You need to set up metrics for them</spoiler>`
https://stackoverflow.com/a/45179856/2089991

**Comment by czytelny** [comment]
Hi, I just want to share with you what cause 'Network Response Time Out Error' error in my case. It wasn't firewall, it wasn't virtual box. My computer with windows which I use to host React Native App was connected to my home network but was unreachable from any other computer (or phone) because it was setup as a "Guest or public" network profile. When I changed this network profile to "Private" it started working :) cheers

**Comment by vishalgadhiya91** [comment]
Can you try after "Windows firewall " Off ?

**Comment by ziggyjosh16** [comment]
My issue probably stems from the fact that the app is running from a Linux Virtual machine. How can I get my device and the virtual machine to be on the same network?

**Comment by harisvsulaiman** [comment]
I have this issue in a production application where expo doesnt connect to ip.
```

arrow_drop_down
Exception java.lang.RuntimeException: java.net.SocketTimeoutException: failed to connect to [REDACTED_DOMAIN_NAME]/[REDACTED_IP_ADDRESS] (port 443) after 10000ms
host.exp.exponent.kernel.ExpoViewKernel.handleError (ExpoViewKernel.java:76)
host.exp.expoview.ExponentActivity$3$1.onError (ExponentActivity.java:195)
host.exp.expoview.Exponent$2.onFailure (Exponent.java:439)
host.exp.exponent.network.ExponentHttpClient$2$1.onFailure (ExponentHttpClient.java:95)
expolib_v1.okhttp3.RealCall$AsyncCall.execute (RealCall.java:142)
expolib_v1.okhttp3.internal.NamedRunnable.run (NamedRunnable.java:32)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
java.lang.Thread.run (Thread.java:818)
arrow_drop_down
Caused by java.net.SocketTimeoutException: failed to connect to [REDACTED_DOMAIN_NAME]/[REDACTED_IP_ADDRESS] (port 443) after 10000ms
libcore.io.IoBridge.connectErrno (IoBridge.java:169)
libcore.io.IoBridge.connect (IoBridge.java:122)
java.net.PlainSocketImpl.connect (PlainSocketImpl.java:183)
java.net.PlainSocketImpl.connect (PlainSocketImpl.java:456)
java.net.Socket.connect (Socket.java:882)
expolib_v1.okhttp3.internal.platform.AndroidPlatform.connectSocket (AndroidPlatform.java:63)
expolib_v1.okhttp3.internal.connection.RealConnection.connectSocket (RealConnection.java:220)
expolib_v1.okhttp3.internal.connection.RealConnection.connect (RealConnection.java:146)
expolib_v1.okhttp3.internal.connection.StreamAllocation.findConnection (StreamAllocation.java:186)
expolib_v1.okhttp3.internal.connection.StreamAllocation.findHealthyConnection (StreamAllocation.java:121)
expolib_v1.okhttp3.internal.connection.StreamAllocation.newStream (StreamAllocation.java:100)
expolib_v1.okhttp3.internal.connection.ConnectInterceptor.intercept (ConnectInterceptor.java:42)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:92)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:67)
expolib_v1.okhttp3.internal.cache.CacheInterceptor.intercept (CacheInterceptor.java:93)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:92)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:67)
expolib_v1.okhttp3.internal.http.BridgeInterceptor.intercept (BridgeInterceptor.java:93)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:92)
expolib_v1.okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept (RetryAndFollowUpInterceptor.java:120)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:92)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:67)
host.exp.exponent.network.ExponentNetwork$3.intercept (ExponentNetwork.java:140)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:92)
expolib_v1.okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:67)
expolib_v1.okhttp3.RealCall.getResponseWithInterceptorChain (RealCall.java:179)
expolib_v1.okhttp3.RealCall$AsyncCall.execute (RealCall.java:129)
expolib_v1.okhttp3.internal.NamedRunnable.run (NamedRunnable.java:32)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
java.lang.Thread.run (Thread.java:818)
```

**Comment by sumethy** [comment]
Hi,

I had to both disable all other network adaptors AND change the network setting to "private" to get Expo working in Windows. Thank @hrastnik and @czytelny .

**Comment by davidRoussov** [comment]
I can confirm disabling the Windows Firewall fixed this issue for me.

**Comment by AurelGit** [comment]
Same problems, and working without firewall.
What i did to fix it, on Windows 10.

1. The famous "set REACT_NATIVE_PACKAGER_HOSTNAME=my-custom-ip-address"
2. Authorize ports 19000, 19001 and 19002
-- https://stackoverflow.com/a/44600431/1443984
3. change the network to "private" (thanks @sumethy)
-- On Windows 10, you can left click on the "wi-fi" icon in system tray > "Properties" > switch to "Private" in the "Network Profile" section.

It fixed this issue for me.

**Comment by CagatayKolus** [comment]
Hi,

I got same problem and i fixed with turning off all Ethernet Adapters except Wi-Fi Ethernet adapter. If you using any adapter such as VirtualBox Adapter, Hamachi Adapter etc. turn off all of them.

When u type "npm start" it will give you new IP adress and verify QR code again and it will works!

**Comment by CostInCode** [comment]
Ubuntu 16.04 same problem, it's fixed just by disabling firewall `sudo ufw disable`

**Comment by hikkithegene** [comment]
Same problem, in my host pc if I use the IP I get: This site can’t be reached
If I use http://localhost:/19000 it shows a json file.

In my phone nothing works, oh man.

**Comment by iamsayantan** [comment]
I had the same issue on Windows 10. Disabling virtual network adapters and allowing the port through windows firewall fixed it. Thanks to @CyrisXD and @anass05 .

**Comment by ToJen** [comment]
@mkmpvtltd1 thanks that worked for me!

**Comment by jalal246** [comment]
@CristhianUNSa  Amazing.. after a couple of days of searching your solution works perfectly THANK YOU SO MUCH!

**Comment by n3gotium** [comment]
If you have tried all of the above solutions and you still can't connect to your computer from your mobile device and **your both devices get connected throw a router**, then it might be that you have **the wireless isolation feature** turned on somewhere in your router's configuration. 

When this feature is turned on, it disallows clients connected to the Wi-Fi network to interact with devices connected to the wired network, nor can they communicate with each other. They can only access the Internet.

You can read more about this feature here: https://www.howtogeek.com/179089/lock-down-your-wi-fi-network-with-your-routers-wireless-isolation-option/

**Comment by holodny** [comment]
I'm developing on a Centos 7 VirtualBox machine.  CyrisXD's advice was the final piece I needed.  Other than that, if you're developing on VirtualBox, I suggest disabling the host ethernet connection and setting up the USB filter on your guest (VM) to "add the USB filter with all fields set to the values of the selected USB device..." for the phone/tablet and maybe also for the Wifi card on your host machine.  My network is a bridged adapter to the Wifi Adapter on the host.

**Comment by eassymo** [comment]
![image](https://user-images.githubusercontent.com/34582073/37221277-f8d2d93c-238e-11e8-8c3f-1010e2563058.png)

**Comment by putrantoinc** [comment]
changing the network to private in windows 10 works for me. thanks @sumethy @AurelGit

**Comment by ChristianHansen93** [comment]
Uninstalling any virtual machine program solved the problem for me.
Seems like it was using the IPV4 from the virtual machine and not from the Lan adapter.

**Comment by JGeraldoLima** [comment]
I only got it following @mkmpvtltd1 stackoverflow link steps. But before, I did all people said in this issue too Lol. So if you're facing this problem, do all them :)

**Comment by vmukunda** [comment]
If you have used create-react-native-app to setup your project, I solved this issue by setting the hostname in the package.json file with in the project. 
 
Package.json 
------------------
 "android": "set REACT_NATIVE_PACKAGER_HOSTNAME=<<your wifi ip address>> && react-native-scripts android",

**Comment by leo2d** [comment]
I have the same problem today on Win 10  and fix using  @CyrisXD  instructions.. Thanks!

**Comment by syedabdulaala** [comment]
I faced same issue, and found that is was due to firewall.
My network was public and in firewall setting Node.js is only allowed to private network.
Allowing it for public solve my problem.

**Comment by AndreiRosu** [comment]
For me, none of the above worked, but the following worked perfectly: access your router settings and disable the "AP Isolation" property and then the router will let connected devices to communicate with each other.

**Comment by assetcorp** [comment]
For me, it was a firewall problem. I was developing in WSL and the ports were being blocked by windows firewall. I simply added a new rule to my firewall that allowed all ports from within WSL and everything worked fine afterwards.

**Comment by hariangr** [comment]
I think this happened because react native use wrong ip... so when expo try to connect... it failed

this happened to me because I have virtual ethernet (Because VBox)...I solved the problem by following Muhammad Hannan's solution from https://stackoverflow.com/questions/47966887/expo-lan-configuration-doesnt-work-for-new-reactnative-project

**Comment by MrMRamzan** [comment]
Only problem is that you are not using "same local internet connection on your mobile and PC"
I have been facing the same problem but after a lot effort i figure it out 
SOLUTION: Use same internet connection on your mobile(on which  using expo app) and Computer (running react native app)

**Comment by MrMRamzan** [comment]
> I did all the above steps still the error i am getting.
> 
> System : windows 10 Home
> Phone : one plus 6
> 
> Any other checks i need to do ?

https://github.com/react-community/create-react-native-app/issues/144#issuecomment-424768154

**Comment by kmturley** [comment]
I just can't fathom how ridiculous it is to pick a non-standard web port, and then expect people to manually open their firewall ports. Why not use ports in the normal range 3000 or 8000?

In my case, I cannot change the rules due to Admin restrictions. But maybe I can pass through an option? Nope you've also hardcoded the ports and not allow them to be configured:
https://github.com/expo/expo-cli/blob/548ac8b32eac86529ff1a4e755e1fa5f47c1ff0e/packages/xdl/src/Project.ts#L2082

You're giving plenty of people reasons to not use this project...

**Comment by natemartins** [comment]
None of the above suggestions and solutions worked for me.

**Comment by IsraelChn** [comment]
**Don't forget to use Wifi on your smartphone as well!!!** glad to help whoever you are ^_^ btw thanks for all the guys above! :)

**Comment by yanuwarishak** [comment]
I stumbled across this issue and none of the aforementioned solutions worked for me, however **disabling firewall for private network worked for me**.
![Capture](https://user-images.githubusercontent.com/26899600/95579862-918ce400-0a60-11eb-85ec-99fbce6392d9.JPG)

**Comment by sami92news** [comment]
And where to write it? cmd/terminal?
I am using ubuntu, I am not sure if its doing anything, is there way to check if the command worked (other than testing the expo which is not working for me as well).

**Comment by harsh-hustler** [comment]
> My solution I found while struggeling with the same problem on another machine (osx sierra), while it was working on my first one (osx sierra) - McAfee was blocking it (which was only installed on the second one); disabled it's Firewall and voilà, it works (ios/android)

Yeah It worked

**Comment by ZangUlrich** [comment]
L have the same problem. To resolve it, l add a new rule in my firewall for the port with the following commands:
* `sudo ufw allow 19000/tcp`   
* `sudo ufw disable`
* `sudo ufw enable` 
Now run `npm start` and it must work with no problem.

**Comment by thinker732** [comment]
https://stackoverflow.com/a/68881495/15562150

maybe my post will help you it's pretty obvious but i find many people who forget about that

**Comment by TheIrishPizzaGuy** [comment]
![Capture](https://user-images.githubusercontent.com/52389249/168491150-b964f7a2-f55d-4864-b2a9-46559cab6ebb.JPG)

4 days for me...
It was weird because the web version of my react-native build on localhost was working, but expo could not connect.
 Allow an app through windows firewall -> make sure both Node.js items have access.

**Comment by gadgetcrawford** [comment]
It's obvious whoever designed this obviously has coding skills, but not how to fix issues as they occur. With so many people having this same issue, you would think they would figure out that maybe they need to fix the connection issues.  Expo is a great concept, but unfortunately the developers are in over their head.  Back to regular ReactNative CLI for me!

**Comment by MahmoudAbdullahAnani** [comment]
1. قم بالنقر على زر "Start" في قائمة البداية.
2. اكتب "WF" في حقل البحث أو "Windows Firewall" واختر "Windows Defender Firewall with Advanced Security".
3. ستظهر نافذة "Windows Defender Firewall with Advanced Security".
4. في القائمة اليسرى، انقر بزر الماوس الأيمن على "Inbound Rules" واختر "New Rule".
5. ستظهر نافذة مع خيارات متعددة. اختر الخيار "Port" واضغط "Next".
6. حدد "TCP" وأدخل رقم المنفذ الذي يستخدمه تطبيقك في الحقل "Specific local ports". ثم اضغط "Next".
7. اختر "Allow the connection" واضغط "Next".
8.  قم بتحديد الشبكات التي ترغب في تطبيق القاعدة عليها. يمكنك تحديد الخيار "Domain", "Private", و "Public" أو اختيار الخيار الذي يناسب حالتك. ثم اضغط "Next".
9. قم بإعطاء القاعدة اسمًا واضغط "Finish" لإنشاء القاعدة.
( اذا كان الPort مثل 192.168.1.177:19000 فعليك كتابة 19000 )
